sdks/java/extensions/sketching/src/main/java/org/apache/beam/sdk/extensions/sketching/frequency/SketchFrequencies.java

```diff
@@ -0,0 +1,357 @@
+     */
+    public CountMinSketchFn withAccuracy(double epsilon, double confidence) {
+      return new CountMinSketchFn(epsilon, confidence, this.seed);
+    }
+
+    @Override public CountMinSketch createAccumulator() {
+      return new CountMinSketch(this.depth, this.width, this.seed);
+    }
+
+    @Override public CountMinSketch addInput(CountMinSketch accumulator, String element) {
+      accumulator.add(element, 1);
+      return accumulator;
+    }
+
+    @Override public CountMinSketch mergeAccumulators(Iterable<CountMinSketch> accumulators) {
+      Iterator<CountMinSketch> it = accumulators.iterator();
+      if (!it.hasNext()) {
+        return new CountMinSketch(seed, width, depth);
+      }
+      CountMinSketch merged = it.next();
+      try {
+        while (it.hasNext()) {
+          merged = CountMinSketch.merge(merged, it.next());
+        }
+      } catch (FrequencyMergeException e) {
+        // Should never happen because all the accumulators created are of the same type.
+        LOG.error(e.getMessage(), e);
+      }
+      return merged;
+    }
+
+    @Override public CountMinSketch extractOutput(CountMinSketch accumulator) {
+      return accumulator;
+    }
+
+    @Override public Coder<CountMinSketch> getAccumulatorCoder(CoderRegistry registry,
+        Coder inputCoder) {
+      return new CountMinSketchCoder();
+    }
+
+    @Override public Coder<CountMinSketch> getDefaultOutputCoder(CoderRegistry registry,
+        Coder inputCoder) throws CannotProvideCoderException {
+      return new CountMinSketchCoder();
+    }
+
+    @Override public CountMinSketch defaultValue() {
+      return new CountMinSketch(1, 1, 1);
+    }
+  }
+
+  static class CountMinSketchCoder extends CustomCoder<CountMinSketch> {
+
+    private static final ByteArrayCoder BYTE_ARRAY_CODER = ByteArrayCoder.of();
+
+    @Override public void encode(CountMinSketch value, OutputStream outStream) throws IOException {
+        if (value == null) {
+          throw new CoderException("cannot encode a null Count-min Sketch");
+        }
+        BYTE_ARRAY_CODER.encode(CountMinSketch.serialize(value), outStream);
+    }
+
+    @Override public CountMinSketch decode(InputStream inStream) throws IOException {
+      return CountMinSketch.deserialize(BYTE_ARRAY_CODER.decode(inStream));
+    }
+
+    @Override public boolean consistentWithEquals() {
+      return false;
+    }
+
+    @Override public boolean isRegisterByteSizeObserverCheap(CountMinSketch value) {
+      return true;
+    }
+
+    @Override protected long getEncodedElementByteSize(CountMinSketch value) throws IOException {
+      if (value == null) {
+        throw new CoderException("cannot encode a null Count-min Sketch");
+      } else {
+        // depth and width as computed in the CountMinSketch constructor from the relative error and
+        // confidence.
+        int width = (int) Math.ceil(2 / value.getRelativeError());
```

Comment:
This is wonky, there's gotta be a better way to get access to depth and width. One easy way that comes to mind is simply send a PR to StreamLib to expose them.

Reply:
I don't understand why this is wonky? I compute depth and width exactly the same way as in Count-Min Sketch constructors. I can still create a PR for that though.

Reply:
I think @jkff is pointing out that it would be best if we didn't duplicate the internal details of StreamLib and instead asked them to expose this functionality.

I wouldn't block merging this in and instead add a TODO to replace it by calling StreamLib's implementation directly when available and link to a JIRA or PR against StreamLib which exposes it.

Reply:
All right I'm going to send a PR to stream-lib and add a Todo in this function then, thanks @lukecwik